com.unity.netcode.gameobjects/Runtime/Messaging/CustomMessageManager.cs

```diff
@@ -52,13 +52,18 @@ internal void InvokeUnnamedMessage(ulong clientId, FastBufferReader reader)
         /// <param name="clientIds">The clients to send to, sends to everyone if null</param>
         /// <param name="messageBuffer">The message stream containing the data</param>
         /// <param name="networkDelivery">The delivery type (QoS) to send data with</param>
-        public void SendUnnamedMessage(List<ulong> clientIds, FastBufferWriter messageBuffer, NetworkDelivery networkDelivery = NetworkDelivery.ReliableSequenced)
+        public void SendUnnamedMessage(IReadOnlyList<ulong> clientIds, FastBufferWriter messageBuffer, NetworkDelivery networkDelivery = NetworkDelivery.ReliableSequenced)
         {
             if (!m_NetworkManager.IsServer)
             {
                 throw new InvalidOperationException("Can not send unnamed messages to multiple users as a client");
             }
 
+            if (clientIds == null)
```

Comment:
would we also want an empty list to mean all clients ?

Reply:
Personally I don't think so. Imagine a case where someone tries to send a message with a LINQ filter on ConnectedClientIds, figuring if there are no matches the message won't get sent. It might not be the most efficient way to do things, but I think it would be surprising for the case of "nothing matches" to have the same result as "everything matches"

Reply:
wouldn't it be better to have a separate method for this? That way it's very clear what the method is doing and what it's capable of; send to a specific client list, or send to everyone.